### PR TITLE
[Mono] Move several related small files

### PR DESCRIPTION
**Pull request by aaronfranke**
Normally when I write my own code I always keep one class per file. However, some of these files are very small (literally just a few lines) and very related and would be better suited combined. @neikeq 

The following files have been combined:

* `NodeExtensions.cs` and `ResourceLoaderExtensions.cs` have been combined into `Extensions.cs`. But not `StringExtensions.cs` as that file is already fairly large and mostly unrelated.

* `IAwaiter.cs` and `IAwaitable.cs` have been combined into `IAwait.cs`.

* `ExportAttritube.cs`, `GodotMethodAttritube.cs`, `RPCAttributes.cs`, `SignalAttribute.cs`, and `ToolAttribute.cs` have all been combined into `Attributes.cs`.

In total, what was 9 files is now 3. Less files makes it easier to find a specific file, and related code being grouped together makes it easier to edit a specific set of code.

This is purely a style change, nothing should change for end users. All I did here was 1. Move around code and files, 2. Tabs -> Spaces, 3. Removed useless comment in `StringExtensions.cs` that I noticed, and 4. Added a comment to `Extensions.cs` so people don't get confused about `StringExtensions.cs` being separate.

EDIT: I should probably ping @mysticfall to get his opinion.

### CODE REVIEW

**Comment by mysticfall** [comment]
Thanks for the head up. I don't see myself to be an experienced C# programmer, so I can only share my personal opinions for what it's worth.

On this particular convention, I don't have any strong opinion so the suggested change looks good to me. Probably, I wouldn't merge all interfaces if it was my own project, since I think interfaces are quite important (maybe not so much in Godot, in general) so I'd rather want them to be exposed at the top level.

**Comment by aaronfranke** [comment]
If desired, I could just have the current file called `IAwait.cs` and have other interfaces be separate, but I think that it's worth keeping `IAwaiter` and `IAwaitable` together.

**Comment by NathanWarden** [comment]
@aaronfranke  I like the idea of your last statement. Interfaces, in general, do seem like something to keep separated, but similar interfaces being combined under a file that describes them seems like a good idea :)

**Comment by aaronfranke** [comment]
I agree with your opinions, and it's now `IAwait.cs`.

**Comment by neikeq** [comment]
I was thinking, what if we place these files in sub-directories instead? It would require some changes in the build script to search recursively, but that's fairly simple.

**Comment by aaronfranke** [comment]
Sub-directories would be cool. We could do `cs_files/math/Vector3.cs` etc. 

But I still think that these things are minor enough to warrant being grouped together into less files.

**Comment by aaronfranke** [comment]
Folders it is then. I'll leave this open until you do that.

**Comment by neikeq** [comment]
As of 7287300, it build search C# files recursively. The `VERSION.txt` file was removed as well, so you don't need to modify it anymore.

File: modules/mono/glue/cs_files/StringExtensions.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//using System;
+// This file contains extensions to C# strings. The "Extensions" folder is different and contains extensions to existing Godot classes.
```
Comment (neikeq) [inline modules/mono/glue/cs_files/StringExtensions.cs]: I don't see the need for this comment.

**Comment by neikeq** [comment]
I would prefer we name files for partial class extensions differently. The `Extensions` suffix is usually used for classes that provide extension methods (like `StringExtensions`). The convention for partial class extensions seems to be `ClassName.something.cs` where `something` describes what the extension methods do, or what problem is solved.
This can be done later in another PR.

**Review by neikeq — APPROVED** [review]

File: modules/mono/glue/cs_files/StringExtensions.cs (hunk 1 above)
Comment (aaronfranke) [reply modules/mono/glue/cs_files/StringExtensions.cs]: I've removed it.

**Pull request merged**